src/main/java/gyro/azure/compute/VirtualMachineResource.java

```diff
@@ -978,7 +1255,18 @@ public void update(GyroUI ui, State state, Resource current, Set<String> changed
     public void delete(GyroUI ui, State state) {
         Azure client = createClient();
 
+        VirtualMachine virtualMachine = client.virtualMachines().getById(getId());
         client.virtualMachines().deleteById(getId());
+
+        if (getDeleteOsDiskOnTerminate()
+                && virtualMachine != null
+                && !"specialized".equals(getVmImageType())) {
+
+            ui.write("\n Deleting Dynamically created OS Disk (%s)", virtualMachine.osDiskId());
+            client.disks().deleteById(virtualMachine.osDiskId());
+        } else {
+            ui.write("\n OS Disk is controlled by Gyro. Skipping deletion of OS disk.");
```

Comment:
If the disk is Gyro controlled then it won't be orphaned which makes mentioning this unnecessary.